packages/frontend/core/src/modules/workspace/properties/schema.ts

```diff
@@ -0,0 +1,101 @@
+import { z } from 'zod';
+
+// ===== workspace-wide page property schema =====
+export const TagOptionSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  color: z.string(),
+});
+
+export type TagOption = z.infer<typeof TagOptionSchema>;
+
+export enum PageSystemPropertyId {
+  Tags = 'tags',
+  Journal = 'journal',
+}
+
+export enum PagePropertyType {
+  String = 'string',
+  Number = 'number',
+  Boolean = 'boolean',
+  Date = 'date',
+  Tags = 'tags',
+}
+
+export const PagePropertyMetaBaseSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  source: z.string(),
+  type: z.string(),
+});
+
+export const PageSystemPropertyMetaBaseSchema =
+  PagePropertyMetaBaseSchema.extend({
+    source: z.literal('system'),
+  });
+
+export const PageCustomPropertyMetaSchema = PagePropertyMetaBaseSchema.extend({
+  source: z.literal('custom'),
+  type: z.nativeEnum(PagePropertyType),
+});
+
+// ====== page info schema ======
+export const PageInfoItemSchema = z.object({
+  id: z.string(), // property id. Maps to PagePropertyMetaSchema.id
+  hidden: z.boolean().optional(),
+  value: z.any(), // corresponds to PagePropertyMetaSchema.type
+});
+
+export const PageInfoJournalItemSchema = PageInfoItemSchema.extend({
+  id: z.literal(PageSystemPropertyId.Journal),
+  value: z.union([z.string(), z.literal(false)]),
+});
+
+export const PageInfoTagsItemSchema = PageInfoItemSchema.extend({
+  id: z.literal(PageSystemPropertyId.Tags),
+  value: z.array(z.string()),
+});
+
+// ====== workspace properties schema ======
+export const WorkspaceFavoriteItemSchema = z.object({
+  id: z.string(),
+  order: z.number(),
+  type: z.enum(['page', 'collection']),
```

Comment:
Is the term defined here as `Favorite` actually supposed to cover both `Favorite` and `Collection`? Wouldn't this cause some ambiguity and lead to confusion?

```ts
adapter.isFavouriated('id', 'collection');
```

Reply:
I haven't thought through, but can you please elaborate more on the ambiguity?
Do you mean that there is a very low chance that the same id os used for two different object?

Reply:
In my opinion, A 'page' can exist in both 'Favorite' and 'Collection', so there are two concepts. However, here 'Favorite' is used to represent both, which makes the naming a bit strange.

Reply:
I think it is not the correct case.

In the new design we will have data like
```ts
const foo: WorkspaceAffineProperties['favouriates'] = {
  pageId_0: {
    id: 'pageId_0',
    order: 0,
    type: 'page',
  },
  collectionId_0: {
    id: 'collectionId_0',
    order: 1,
    type: 'collection',
  },
};
```
page `pageId_0` could exist in `collectionId_0`, but it is not relevant to the this cases. The page -> collection mapping is handled elsewhere